components/SingleClassView/Attendance/AttendanceComponent.tsx

```diff
@@ -33,13 +39,22 @@ export const AttendanceComponent: React.FC<AttendanceComponentProps> = ({ classI
       );
       if (sessions.length == 0) {
         setAttendance([]);
+        setStudentAttendance([]);
         setLoading(false);
         return <p>This date has no sessions</p>;
       }
       setSessionID(sessions[0].sessionId);
       const sessionId = sessions[0].sessionId;
-      const attendances = await api.getAttendanceFromSessionID(sessionId, classId);
-      setAttendance(attendances);
+      if (user){
+        if (user.role == "Teacher" || user.role == "Admin"){
+          const attendances = await api.getAttendanceFromSessionID(sessionId, classId);
+          setAttendance(attendances);
+        }
+        else{
```

Comment:
commenting for why we are splitting this up based on roles

Reply:
The Attendance objects returned are different. The Teachers and Admin get a list of Attendance objects whereas a student gets a list of SingleUserAttendance objects